testing/trino-tests/src/test/java/io/trino/memory/TestMemoryManager.java

```diff
@@ -324,7 +324,9 @@ public void testClusterPools()
             for (TestingTrinoServer worker : queryRunner.getServers()) {
                 Optional<MemoryPool> reserved = worker.getLocalMemoryManager().getReservedPool();
                 assertTrue(reserved.isPresent());
-                assertEquals(reserved.get().getMaxBytes(), reserved.get().getFreeBytes());
+                while (reserved.get().getMaxBytes() != reserved.get().getFreeBytes()) {
+                    MILLISECONDS.sleep(10);
```

Comment:
`assertEventually`?

also, explain why eventual consistency can be observed here